extensions/wikia/AbPerfTesting/js/FrontendDelay.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Introduce a delayed onDOMReady event to measure the impact of performance regressions
+ *
+  * @see FrontendDelay.class.php
+ */
+(function(w, log, weppy, $) {
+
+	function updateTransactionName(perf_test) {
+		// update Weppy config when transaction context is updated
+		// matches \Transaction::PARAM_AB_PERFORMANCE_TEST const
+		w.wgTransactionContext['perf_test'] = perf_test;
+
+		weppy.setOptions({
+			context: w.wgTransactionContext
+		});
+
+		// set UA custom dimension
+		// should match the PHP code in /extensions/wikia/AbPerfTesting/classes/Hooks.class.php
+		if (perf_test) {
+			w._gaq.push(
```

Comment:
I do agree, I'd like to keep the logic in a single place - i.e. in the backend code. However, CDN caching of front-end stuff does not allow us to do so.

Front-end delays (if handled by the backend script) for the testing group (chosen by modulo of `beacon_id`) would pollute the cache for clients from outside the test group.